### ISSUE TITLE
Docker custom networks only respect first `--dns` flag in `docker run`

### ISSUE THREAD

**Issue by shane-axiom** [docker/libnetwork#1287]
Using Docker 1.11.2. These docs:

https://docs.docker.com/engine/userguide/networking/configure-dns/

indicate that the internal DNS server on a custom network should respect `--dns` flags when running `docker run` and use those as external DNS servers when the internal DNS can't resolve a host.

However, it appears that only the first `--dns` flag is respected. Passing two results in the second `--dns` being ignored, and reversing the order of the `--dns` flags causes reversed results.

Please see this gist for code to reliably replicate this behavior (I can inline this script if preferred):

https://gist.github.com/shane-axiom/9773e9fbaf6496f7dbe30168388ee328

**Comment by shane-axiom** [comment]
Yes, this issue is still relevant. I just tested again against docker 17.05.0-ce and updated the gist demonstrating the issue:

https://gist.github.com/shane-axiom/9773e9fbaf6496f7dbe30168388ee328

**Comment by stefan-wegener** [comment]
Is there still something happening regarding this issue? For me Docker 18.05 is still facing the above problems. Btw, I think the issue is related to: [https://github.com/docker/libnetwork/issues/1664](https://github.com/docker/libnetwork/issues/1664)
